arch/sim/include/inttypes.h

```diff
@@ -44,7 +44,7 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-#if defined(CONFIG_HOST_MACOS)
+#if defined(CONFIG_HOST_MACOS) && defined(__APPLE_CC__) || !defined(_LP64)
```

Comment:
is it enough to only check __APPLE_CC__?
```
#if defined(__APPLE_CC__) || !defined(_LP64)
```

Reply:
no. it's for MODULECC. (see the commit message)

Reply:
If I undersand correctly, macOS has two cases:

1. Compile the image with clang(__INT64_TYPE__ == long long)
2. Compile the module with gcc(__INT64_TYPE__ == long)

In the first case, both CONFIG_HOST_MACOS and __APPLE_CC__ are defined
In the second case, CONFIG_HOST_MACOS is defined, but __APPLE_CC__ isn't.
So the following definition should be good:
```
#if defined(__APPLE_CC__)
 typedef signed long long   _int64_t;
 typedef unsigned long long _uint64_t;
#else
 typedef signed long        _int64_t;
 typedef unsigned long      _uint64_t;
#endif
```
If we consider Linux 32bit mode(__INT64_TYPE__ == long long), it become:
```
#if defined(__APPLE_CC__) || !defined(_LP64)
 typedef signed long long   _int64_t;
 typedef unsigned long long _uint64_t;
#else
 typedef signed long        _int64_t;
 typedef unsigned long      _uint64_t;
#endif
```

Reply:
done